[2024-06-10T07:57:54.596+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-10T07:57:54.612+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: f1_data_pipeline_taskflow_with_kafka.consume_and_load_driverDim manual__2024-06-10T07:57:50.563407+00:00 [queued]>
[2024-06-10T07:57:54.617+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: f1_data_pipeline_taskflow_with_kafka.consume_and_load_driverDim manual__2024-06-10T07:57:50.563407+00:00 [queued]>
[2024-06-10T07:57:54.617+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-06-10T07:57:54.625+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): consume_and_load_driverDim> on 2024-06-10 07:57:50.563407+00:00
[2024-06-10T07:57:54.630+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=822) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-10T07:57:54.630+0000] {standard_task_runner.py:63} INFO - Started process 824 to run task
[2024-06-10T07:57:54.631+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'f1_data_pipeline_taskflow_with_kafka', 'consume_and_load_driverDim', 'manual__2024-06-10T07:57:50.563407+00:00', '--job-id', '42', '--raw', '--subdir', 'DAGS_FOLDER/kafkaProduce.py', '--cfg-path', '/tmp/tmp7y7sgm_3']
[2024-06-10T07:57:54.633+0000] {standard_task_runner.py:91} INFO - Job 42: Subtask consume_and_load_driverDim
[2024-06-10T07:57:54.664+0000] {task_command.py:426} INFO - Running <TaskInstance: f1_data_pipeline_taskflow_with_kafka.consume_and_load_driverDim manual__2024-06-10T07:57:50.563407+00:00 [running]> on host e8457e1e96c0
[2024-06-10T07:57:54.728+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='f1_data_pipeline_taskflow_with_kafka' AIRFLOW_CTX_TASK_ID='consume_and_load_driverDim' AIRFLOW_CTX_EXECUTION_DATE='2024-06-10T07:57:50.563407+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-10T07:57:50.563407+00:00'
[2024-06-10T07:57:54.729+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-10T07:57:54.731+0000] {conn.py:362} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.6', 9092)]>: connecting to kafka:9092 [('172.18.0.6', 9092) IPv4]
[2024-06-10T07:57:54.732+0000] {conn.py:957} INFO - Probing node bootstrap-0 broker version
[2024-06-10T07:57:54.732+0000] {conn.py:393} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.6', 9092)]>: Connection complete.
[2024-06-10T07:57:54.836+0000] {conn.py:1019} INFO - Broker version identified as 2.6.0
[2024-06-10T07:57:54.837+0000] {conn.py:1020} INFO - Set configuration api_version=(2, 6, 0) to skip auto check_version requests on startup
[2024-06-10T07:57:54.838+0000] {consumer.py:122} WARNING - group_id is None: disabling auto-commit.
[2024-06-10T07:57:54.838+0000] {subscription_state.py:167} INFO - Updating subscribed topics to: ('f1_data_topic_driver',)
[2024-06-10T07:57:54.840+0000] {crypto.py:82} WARNING - empty cryptography key - values will not be stored encrypted.
[2024-06-10T07:57:54.840+0000] {base.py:84} INFO - Using connection ID 'sourcedb_connection' for task execution.
[2024-06-10T07:57:54.843+0000] {subscription_state.py:253} INFO - Updated partition assignment: [('f1_data_topic_driver', 0)]
[2024-06-10T07:57:54.845+0000] {conn.py:362} INFO - <BrokerConnection node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.6', 9092)]>: connecting to kafka:9092 [('172.18.0.6', 9092) IPv4]
[2024-06-10T07:57:54.845+0000] {conn.py:393} INFO - <BrokerConnection node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.6', 9092)]>: Connection complete.
[2024-06-10T07:57:54.846+0000] {conn.py:673} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.6', 9092)]>: Closing connection. 
[2024-06-10T07:57:55.007+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-10T07:57:55.007+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/kafkaProduce.py", line 125, in consume_and_load_driverDim
    drivers = data['MRData']['DriverTable']['Drivers']
              ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
KeyError: 'DriverTable'
[2024-06-10T07:57:55.017+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=f1_data_pipeline_taskflow_with_kafka, task_id=consume_and_load_driverDim, run_id=manual__2024-06-10T07:57:50.563407+00:00, execution_date=20240610T075750, start_date=20240610T075754, end_date=20240610T075755
[2024-06-10T07:57:55.025+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 42 for task consume_and_load_driverDim ('DriverTable'; 824)
[2024-06-10T07:57:55.044+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-10T07:57:55.056+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-10T07:57:55.057+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
